### ISSUE TITLE
Differential Privacy with Varying Epsilon Values in parameter:  No Impact on Output

### ISSUE THREAD

**Issue by Hafizamariaiqbal** [pytorch/opacus#570]
## 🐛 Bug

Using Differential privacy code , **hypermeter (eps)** value change do not effect the result.

# initialize clients numbers
client_num = 10
# load data
d = load_mnist(client_num)

lr = 0.01
fl_param = {
    'output_size': 10,              # number of units in output layer
    'client_num': client_num,       # number of clients
    'C': 1,
    'model': MLP,                   # model
    'data': d,                      # dataset
    'q': 0.1,                       # sampling rate
    'tot_T': 100,                    # number of aggregation times
    'E': 5,                        # number of local iterations
    'batch_size': 64,
    'lr': 0.01,                      # learning rate
    'clip': 1,                      # clipping norm
    'eps': 0.5,                      # privacy budget for each global communication

    # 'eps': 50,                      # privacy budget for each global communication
    'delta': 1e-5,                  # approximate differential privacy: (epsilon, delta)-DP
    'device': device
}
import warnings
warnings.filterwarnings("ignore")
start_time = time.time()
fl_entity = FLServer(fl_param).to(device)
print('Currently performing FL with DP ---------------------------:')
print(datetime.datetime.now().strftime("%A, %d. %B %Y %I:%M%p"))
print(f'CPUs: {torch.get_num_threads()}, GPUs: {torch.cuda.device_count()} on {socket.gethostname()}.')
acc = []
loss = []
for t in range(fl_param['tot_T']):
    fl_entity.set_lr(lr)
    ''' Update the local model according to weights update '''
    loss_, acc_ = fl_entity.global_update()
    loss += [loss_]
    acc += [acc_]
    print("Round = {:d}, loss={:.4f}, acc = {:.4f}".format(t+1, loss[-1], acc[-1]))
print(f'Total time: {time.time() - start_time :.2f} s.')

**Comment by alexandresablayrolles** [comment]
Can you please share more information, like the test accuracy you obtain at various levels of epsilon?

**Comment by Hafizamariaiqbal** [comment]
> Can you please share more information, like the test accuracy you obtain at various epsilon levels?

Yes, first I execute code with epsilon value 0.5 with different noise added. In the next experiment, I did with the same parameter but change the epsilon value to 10, but no change after that I increase the value to 50 but the same results and epsilon in each round.  No accuracy increases or decreases with the change of epsilon.

**Comment by Victor-QTP** [comment]
I encountered this problem too.
For various epsilon values, the accuracy remains pretty much the same

**Target epsilon = 50**
============ Train epoch 0 ============                                                     
python3.10/site-packages/torch/nn/modules/module.py:1344: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "                                                                                     
 Site-1         | Train Loss: 0.6462 | Train Acc: 0.6173|ε =32.90814469011547|δ =4.4764761179999104e-05|nm=0.2766227722167969                                                           
 Site-2         | Train Loss: 0.6078 | Train Acc: 0.6941|ε =32.90814469011547|δ =4.4764761179999104e-05|nm=0.2766227722167969                                                           
 Site-3         | Train Loss: 0.6603 | Train Acc: 0.5765|ε =32.90814469011547|δ =4.4764761179999104e-05|nm=0.2766227722167969                                                           
 Site-4         | Train Loss: 0.5880 | Train Acc: 0.7240|ε =32.90814469011547|δ =4.4764761179999104e-05|nm=0.2766227722167969                                                           
 Site-5         | Train Loss: 0.6630 | Train Acc: 0.5800|ε =32.90814469011547|δ =4.4764761179999104e-05|nm=0.2766227722167969

**Target epsilon =0.5**
============ Train epoch 0 ============                                                                                                                                                 python3.10/site-packages/torch/nn/modules/module.py:1344: UserWarning: Using a non-full backward hook when the forward contains multiple autograd Nodes is deprecated and will be removed in future versions. This hook will be missing some grad_input. Please use register_full_backward_hook to get the documented behavior.
  warnings.warn("Using a non-full backward hook when the forward contains multiple autograd Nodes "                                                                                     
 Site-1         | Train Loss: 0.6480 | Train Acc: 0.6133|ε =0.24494938113499243|δ =4.4764761179999104e-05|nm=1.689453125                                                                
 Site-2         | Train Loss: 0.6078 | Train Acc: 0.6945|ε =0.24494938113499243|δ =4.4764761179999104e-05|nm=1.689453125                                                                
 Site-3         | Train Loss: 0.6592 | Train Acc: 0.5773|ε =0.24494938113499243|δ =4.4764761179999104e-05|nm=1.689453125                                                                
 Site-4         | Train Loss: 0.5885 | Train Acc: 0.7229|ε =0.24494938113499243|δ =4.4764761179999104e-05|nm=1.689453125                                                                
 Site-5         | Train Loss: 0.6635 | Train Acc: 0.5743|ε =0.24494938113499243|δ =4.4764761179999104e-05|nm=1.689453125

**Comment by HuanyuZhang** [comment]
As an immediate step, could you print out the value of noise_multiplier (https://github.com/pytorch/opacus/blob/main/opacus/privacy_engine.py#L475)? That is the amount of noise added to the system. Sometimes when the noise difference is small, the model is robust to the noise change.
